### DISCUSSION TITLE
关于matrix操作

### DISCUSSION THREAD

**Discussion by 0x1306a94** [Tencent/libpag#1885] [General] [answered]
比如我动态调整了文本图层的内容后，没有自动居中了。所以我通过以下方式调整matrix来实现居中。
当我使用第一种，发现不符合预期。从现象来看，resetMatrix 是将其恢复到AE里面设计的初始位置。
当我使用第二种方式时，可以达到效果。那么有没有方式可以将某个layer直接重置到父级的原点
```C++
do {
        auto textRect = m_player->getBounds(target);
        auto matrix = pag::Matrix::I();
#if 0
        matrix.postTranslate(m_file->width() * 0.5, m_file->height() * 0.5);
        target->resetMatrix();
#else
        auto tx = m_file->width() * 0.5 - textRect.centerX();
        auto ty = m_file->height() * 0.5 - textRect.centerY();
        matrix.postTranslate(tx, ty);
#endif
        target->setMatrix(matrix);
} while (0);
```

[pag_cpp_test.zip](https://github.com/Tencent/libpag/files/13386896/pag_cpp_test.zip)

**Comment by kevingpqi123** [comment, marked as answer]
都是需要去计算位置信息的，然后根据具体情况做偏移，这是 Matrix 相关的基础操作，没有什么捷径

**Comment by 0x1306a94** [comment]
@kevingpqi123 我知道是需要通过 Matrix 去实现，比如上面这个示例中执行 target->resetMatrix() 是指重置到AE设计时的位置对吧。
另外通过 Matrix 调整位置，是以左上角为原点吗？

**Reply by kevingpqi123** [reply]
是的，基本上所有的图形绘制都是这样的逻辑，不分平台

**Comment by 0x1306a94** [comment]
@kevingpqi123 还有个问题想请教下，`pag::PAGFont::SetFallbackFontPaths` 第二个参数的 `ttc`值是怎么获取的。我查了下好像是`Linux`平台才需要这个，`Apple`平台不需要
